dali/operators/generic/slice/slice_attr.h

```diff
@@ -53,11 +56,17 @@ class SliceAttr {
     // slice args are CPU inputs
     const auto &crop_anchor = ws.template Input<CPUBackend>(1);
     const auto &crop_shape = ws.template Input<CPUBackend>(2);
+    DALI_ENFORCE(crop_anchor.type().id() == crop_shape.type().id(),
+                  make_string("Anchor and shape should have the same type. Got: ",
+                              crop_anchor.type().id(), " and ", crop_shape.type().id()));
+    auto args_dtype = crop_anchor.type().id();
     for (std::size_t data_idx = 0; data_idx < batch_size__; data_idx++) {
       VerifyArgsShape(crop_anchor.tensor_shape(data_idx), crop_shape.tensor_shape(data_idx));
-      ProcessArgumentsHelper(data_idx,
-                             crop_anchor.template tensor<float>(data_idx),
-                             crop_shape.template tensor<float>(data_idx));
+      TYPE_SWITCH(args_dtype, type2id, ArgsType, SLICE_ARGS_TYPES, (
```

Comment:
I think the loop over samples should be inside the type switch - the type can't change from sample to sample.